blocks/core/ff_module/ff_module-task-event/_src/taskEventWithOptionalMessage.js

```diff
@@ -1,17 +1,67 @@
 'use strict';
 var React = require('react/addons');
 var TaskEventBase = require('./TaskEventBase.js');
+var eventStates = require('./events').states;
 
 
-module.exports = function(text) {
+function renderEdited(){
+    var messageText = this.props.description.message || '',
+        outputText = getText('stamped response as seen.', this.props),
+        editedFlag = getEditedFlag(this.props);
 
+        var message = messageText ? <p className="ff_module-task-event__message">{messageText}{editedFlag}</p> : null;
+
+        return  <TaskEventBase
+                    description={this.props.description}
+                    actions={this.props.actions}
+                    state={this.props.state}
+                    onRetryAfterStatusError={this.props.onRetryAfterStatusError}>
+                    <p className='ff_module-task-event__author-action'>{outputText}</p>
+                    {message}
+                </TaskEventBase>;
+}
+
+function defaultState(text) {
     return function() {
         var messageText = this.props.description.message || '',
             outputText = getText(text, this.props);
 
         var message = messageText ? <p className="ff_module-task-event__message">{messageText}</p> : null;
 
-        return  <TaskEventBase description={this.props.description}>
+        return  <TaskEventBase
+                    description={this.props.description}
+                    actions={this.props.actions}
+                    state={this.props.state}
+                    onRetryAfterStatusError={this.props.onRetryAfterStatusError}>
+                    <p className='ff_module-task-event__author-action'>{outputText}</p>
+                    {message}
+                </TaskEventBase>;
```

Comment:
Some duplication here between the rendering of the different states. Could extract this out into a function that takes `props`, `outputText` and `message` as parameters.